sklearn/decomposition/_nmf.py

```diff
@@ -1188,6 +1215,24 @@ def _check_params(self, X):
     def _check_w_h(self, X, W, H, update_H):
         """Check W and H, or initialize them."""
         n_samples, n_features = X.shape
+        if self.n_components == "auto":
```

Comment:
```suggestion
        if self._n_components == "auto":
```

Reply:
I think here we shouldn't replace it by _n_components here as _n_components is set to X.shape[1] a few lines before, in _check_params, when "auto" is selected. So that statement would never be true